pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -455,23 +455,17 @@ public synchronized void producerRemoved(String producerName) {
     public synchronized void purgeInactiveProducers() {
```

Comment:
do we need to keep this 'synchronized' ?

Reply:
I think we don't need it, and I also want to remove it in this PR initially. But considering it is not related to the publish latency spike issue, maybe we'd better just keep this PR only for the publish latency spike issue. And I can create the following PR to remove this `synchronized`, but no need to cherry-pick to release branches.